src/index.js

```diff
@@ -39,7 +39,14 @@ if (
 }
 
 /* Warning if there are several instances of styled-components */
-if (process.env.NODE_ENV !== 'production' && typeof window !== 'undefined') {
+if (
+  process.env.NODE_ENV !== 'production' &&
```

Comment:
I think we should merge this check into the `IS_BROWSER` constant: https://github.com/styled-components/styled-components/blob/master/src/constants.js#L11-L12

That'd probably fix a lot of issues for ppl who're running in jsdom. On the other hand, they'll lose access to style tags in jsdom... Quite the weird issue.

Reply:
Let's evaluate for 4.0? It'd be a breaking change.